File: BehaviorPattern/Memento/Golang/Helpers/caretaker.go
package helpers

// Caretaker is responsible for managing Mementos
type Caretaker struct {
	mementoList []*Memento
}

// Add adds a Memento to the list
func (c *Caretaker) Add(m *Memento) {
	c.mementoList = append(c.mementoList, m)
}

// Get retrieves a Memento from the list by index
func (c *Caretaker) Get(index int) *Memento {
	if index < 0 || index >= len(c.mementoList) {
		return nil
	}
	return c.mementoList[index]
}

File: BehaviorPattern/Memento/Python/Helpers/Caretaker.py
class Caretaker:
    def __init__(self):
        """
        Initializes the Caretaker with an empty list of Mementos.
        """
        self._memento_list = []

    def add(self, memento):
        """
        Adds a Memento to the list.
        :param memento: The Memento to add.
        """
        self._memento_list.append(memento)

    def get(self, index):
        """
        Retrieves a Memento from the list.
        :param index: The index of the Memento to retrieve.
        :return: The retrieved Memento.
        """
        return self._memento_list[index]
